src/Sarif/Core/StackFrame.cs

```diff
@@ -0,0 +1,160 @@
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// A stack frame of a SARIF stack.
+    /// </summary>
+    public partial class StackFrame
+    {
+        internal const string IN = " in ";
+        internal const string AT = "   at ";
+        internal const string LINE = ":line";
+
+        /// <summary>
+        /// Creates a SARIF StackFrame instance from a .NET StackFrame instance
+        /// </summary>
+        /// <param name="stackTrace"></param>
+        /// <returns></returns>
+        public static StackFrame Create(System.Diagnostics.StackFrame dotNetStackFrame)
+        {
+            // This value is -1 if not present
+            string fileName = dotNetStackFrame.GetFileName();
+            int ilOffset = dotNetStackFrame.GetILOffset();
+            int nativeOffset = dotNetStackFrame.GetNativeOffset();
+            MethodBase methodBase = dotNetStackFrame.GetMethod();
+            Assembly assembly = methodBase?.DeclaringType.Assembly;
+            string fullyQualifiedName = CreateFullyQualifiedName(methodBase);
+
+            PhysicalLocation physicalLocation;
+            
+            if (fileName != null)
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Uri = new Uri(fileName),
+                    Region = new Region()
+                    {
+                        StartLine = dotNetStackFrame.GetFileLineNumber(),
+                        StartColumn = dotNetStackFrame.GetFileColumnNumber(),
+                        Offset = ilOffset
+                    }
+                };
+            }
+            else
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Region = new Region() { Offset = ilOffset }
+                };
+            }
+
+            StackFrame stackFrame = new StackFrame()
+            {
+                FullyQualifiedLogicalName = fullyQualifiedName,
+                Location = physicalLocation,
+                Module = assembly?.GetName().Name,                 
+            };
+
+            if (nativeOffset != -1)
+            {
+                stackFrame.Properties = new Dictionary<string, string>
+                {
+                    { "NativeOffset", nativeOffset.ToString(CultureInfo.InvariantCulture) }
+                };
+            }
+
+            return stackFrame;
+        }
+
+        public override string ToString()
+        {
+            string result = AT + this.FullyQualifiedLogicalName;
+
+            if (Location?.Uri != null)
+            {
+                string lineNumber = Location.Region.StartLine.ToString(CultureInfo.InvariantCulture);
+                string fileName = Location.Uri.LocalPath;
+
+                result += IN + fileName + LINE + " " + lineNumber;
+            }
+
+            return result;
+        }
+
+        public static string CreateFullyQualifiedName(MethodBase methodBase)
```

Comment:
I agree that _if_ it were public, it doesn't belong here. More likely an extension method on `MethodBase`. But I don't think it needs to be public.